dev/http_test.go

```diff
@@ -0,0 +1,51 @@
+package dev
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var testHttp HTTPServer
+
+func TestHttp_removeTLD_test(t *testing.T) {
+	str := testHttp.removeTLD("psychic-octo-guide.test")
+
+	assert.Equal(t, "psychic-octo-guide", str)
+}
+
+func TestHttp_removeTLD_noTld(t *testing.T) {
+	str := testHttp.removeTLD("shiny-train")
+
+	assert.Equal(t, "shiny-train", str)
+}
+
+func TestHttp_removeTLD_mutlipartDomain(t *testing.T) {
+	str := testHttp.removeTLD("expert-eureka.loc.al")
+
+	assert.Equal(t, "expert-eureka.loc", str)
+}
+
+func TestHttp_removeTLD_dev(t *testing.T) {
+	str := testHttp.removeTLD("bookish-giggle.dev:8080")
+
+	assert.Equal(t, "bookish-giggle", str)
+}
+
+func TestHttp_removeTLD_xipIoMalformed(t *testing.T) {
+	str := testHttp.removeTLD("legendary-meme.0.0.xip.io")
+
+	assert.Equal(t, "", str)
```

Comment:
this is results in "unknown app" error, since "" doesn't resolve to an existing app.